tests/test_template_commodity_macros.py

```diff
@@ -0,0 +1,24 @@
+# pylint: disable=missing-docstring
+from beancount.core.data import Transaction
+from flask import g
+from flask import get_template_attribute
+
+
+COMMODITY_MACROS_PATH = "macros/_commodity_macros.html"
+
+
+def test_render_currency(app, example_ledger) -> None:
+    with app.test_request_context(""):
+        macro = get_template_attribute(
+            COMMODITY_MACROS_PATH, "render_currency"
+        )
+        assert "US Dollar" in macro(example_ledger, "USD")
+
+
+def test_render_amount(app, example_ledger) -> None:
+    with app.test_request_context(""):
+        macro = get_template_attribute(COMMODITY_MACROS_PATH, "render_amount")
+        g.ledger = example_ledger
```

Comment:
This is a bit dirty as some filters are depending on `g.ledger`. Is it OK to leave it "AS IS"?

Reply:
Using `app.preprocess_request()` here should do the trick. Will do so in the merge